client/src/components/MapAllLocationMarker.jsx

```diff
@@ -6,69 +6,67 @@ export default function AllLocationMarker({
   locations,
   roadtrip,
   onUpdateRoadtripLocations,
+  showAllLocations,
 }) {
-  const addLocation = (event) => {
-    const index = locations.findIndex(
-      (oneLocation) => oneLocation._id === event.target.id
-    );
+  const addLocation = (index) => {
     const newLocation = locations[index];
     let updatedRoadtripLocations;
     if (
-      roadtrip?.roadtripLocations.some(
+      roadtrip?.locations?.some(
         (element) => element._id === locations[index]._id
       )
     ) {
       return null;
     } else {
-      updatedRoadtripLocations = [...roadtrip.roadtripLocations, newLocation];
+      updatedRoadtripLocations = [...roadtrip.locations, newLocation];
     }
     onUpdateRoadtripLocations({
       ...roadtrip,
-      roadtripLocations: updatedRoadtripLocations,
+      locations: updatedRoadtripLocations,
     });
-    console.log(roadtrip);
   };
 
   return (
     <>
-      {locations
-        .filter((oneLocation) => {
-          return !roadtrip.roadtripLocations.some((oneRoadtriplocation) => {
-            return (
-              oneRoadtriplocation.properties.name ===
-              oneLocation.properties.name
-            );
-          });
-        })
-        .map((oneLocation, index) => (
-          <Marker
-            key={oneLocation._id}
-            position={[
-              oneLocation.geometry.coordinates[1],
-              oneLocation.geometry.coordinates[0],
-            ]}
-            icon={getIcon(oneLocation.type)}>
-            <Popup offset={[0, -5]}>
-              <PopupContent>
-                <h3>{oneLocation.properties.name}</h3>
-                {oneLocation.properties.website && (
-                  <>
-                    <p>
-                      Visit{" "}
-                      <a href={oneLocation.properties.website} target='_blank'>
-                        website
-                      </a>{" "}
-                      for more details!
-                    </p>
-                  </>
-                )}
-                <button onClick={addLocation} id={oneLocation._id} name={index}>
-                  +
-                </button>
-              </PopupContent>
-            </Popup>
-          </Marker>
-        ))}
+      {showAllLocations &&
+        locations
+          .filter((oneLocation) => {
+            return !roadtrip?.locations?.some((oneRoadtriplocation) => {
+              return (
+                oneRoadtriplocation.properties.name ===
+                oneLocation.properties.name
+              );
+            });
+          })
```

Comment:
This logic could be extracted into its own function
```suggestion
          .filter((location) => !isLocationPartOfRoadtrip(location, roadtrip))
```

Reply:
The function could look as following:
```js
const isLocationPartOfRoadtrip = (location, roadtrip) =>
    roadtrip?.locations?.some(
      (roadtripLocation) =>
        roadtripLocation.properties.name === location.properties.name
    );
```